src/services/jsDoc.ts

```diff
@@ -68,6 +68,28 @@ namespace ts.JsDoc {
         return documentationComment;
     }
 
+    export function getJsDocTagsFromDeclarations(declarations: Declaration[]) {
+        // Only collect doc comments from duplicate declarations once.
+        const tags: JSDocTagInfo[] = [];

```

Comment:
Here too the array may be lazy-allocated when first tag is encountered -- meaning no allocation for the default case.